src/sources/journald.rs

```diff
@@ -180,6 +190,28 @@ impl JournaldConfig {
         }
         matches
     }
+
+    fn schema_definition(&self, log_namespace: LogNamespace) -> Definition {
```

Comment:
This appears to be missing the `*` from https://vector.dev/docs/reference/configuration/sources/journald/#event-fields

Reply:
:eyes: def haven't been referencing the docs :grimacing: 
That's good to know!

Reply:
How to add that to the schema is actually a little puzzling :thinking: 
It looks like as we de serialize the journalctl messages, we create the log event from the key-value pairs of the JSON, rather than inserting new fields into an existing log.

We could do something like this I guess?

```Rust
//...
            .with_source_metadata(
                JournaldConfig::NAME,
                None,
                &owned_value_path!("*"),
                Kind::object(Collection::empty().with_unknown(Kind::bytes())).or_undefined(),
                None,
            )
```

But regardless of how we define it in the schema, I think I may have to change this source from creating the event from the `Record` , to creating an empty log event and looping over the Record elements and adding them to the event for the appropriate namespace.

Reply:
I believe the `*` isn't a literal `*` but means 'all the other fields that are in the journald record'.

I don't think it is correct to use the `BytesDeserializerConfig` schema definition here. That makes the root field a String for the Vector namespace, where it should really be an object containing  some fields that we don't know about at boot time.

Reply:
Indeed, this should probably be `JsonDeserializerConfig`

Reply:
> I believe the * isn't a literal * but means 'all the other fields that are in the journald record'.

Right, I got that. But I'm just wondering how to represent that in the schema. If we don't store it in a collection,  how else should we represent that?
The other sources I've done, we always stored arbitrary lists of fields in a collection.

Reply:
Or is the suggestion to rename `"` to something more meaningful like "journald_record" ?

Reply:
@neuronull I think the crux of the issue is that you're using the `BytesDeserializer` when it should be a `JsonDeserializer`.

Reply:
As far as I can tell from the code there is not absolutely guaranteed to be a message field.

So the schema definition for the Vector namespace should essentially something similar to:

```rust
schema::Definition::new_with_default_metadata(
  Kind::object(Collection::empty()
                         .with_unknown(Kind::bytes())
                         .with_known("message", Kind::bytes().or_undefined())
)), [log_namespace]);
```

And I think Legacy should be on the same lines.

I think using the schema definition from `BytesDeserializerConfig` is not the right way to do it since that says there is definitely a message field.

Reply:
I'm not 100% sure we need the `with_known` there.

Reply:
> when it should be a `JsonDeserializer`.

I don't think it should be `JsonDeserializer` since that says all the fields have to be a Json type (string, int etc..). We are creating the Event from:

```rust
type Record = HashMap<String, String>;
```

So we know all the fields will be `String`.

Reply:
> I'm not 100% sure we need the with_known there since that isn't really adding any extra information that the with_unknown is giving us.

I see your point. If they are behaviorally the same, I sort of prefer having the `with_known` because it's more explicit to someone reading the schema definition logic, that we expect a `message` field but it might not be added.

Reply:
Resolved in https://github.com/vectordotdev/vector/pull/15326/commits/89b75e0f08657ac235d73228c556000affea90c4 and https://github.com/vectordotdev/vector/pull/15326/commits/75093e3477c8f8190a2a2813924798699176ccde